### PR TITLE
RUN-130: Allowed apostrophe character on first name and last name fields

### PR DESCRIPTION
**Pull request by alexander-variacode**
Issue:
https://github.com/rundeckpro/rundeckpro/issues/1720

Fix description:
Allowed apostrophe character on first name and last name fields.

### CODE REVIEW

File: rundeckapp/grails-app/domain/rundeck/User.groovy [hunk 1]
```diff
@@ -41,8 +41,8 @@ class User {
     static hasMany = [reportfilters:ReportFilter,jobfilters:ScheduledExecutionFilter,nodefilters:NodeFilter]
     static constraints={
         login(matches: '^[a-zA-Z0-9\\.,@\\(\\)\\s_\\\\/-]+$')
```
Comment (qualman) [inline rundeckapp/grails-app/domain/rundeck/User.groovy:43]: Did we want to expand this regex to include `\\p{L}\\p{M}` as well, so as to include all UTF-8 alpha-numeric characters, and/or just remove the validation? That way we address both #1720 and #1806?

Comment (jtobard) [reply rundeckapp/grails-app/domain/rundeck/User.groovy:43]: In my opinion, if we have clients that are using characters outside that regex that come from external integrations, as is the case rundeckpro/rundeckpro#1806 we should leave this field with the new regex that is more universal expecting more users using different alphabets.

Comment (qualman) [reply rundeckapp/grails-app/domain/rundeck/User.groovy:43]: @fdevans , how do you feel about expanding this regex to something like `login(matches: '^[a-zA-Z0-9\\p{L}\\p{M}\\.,@\\(\\)\\s_\\\\/-]+$')` (suggested by @alexander-variacode) in order to address rundeckpro/rundeckpro#1806 and allow users with additional character sets to use their normal alphabet as a login name?

Comment (gschueler) [reply rundeckapp/grails-app/domain/rundeck/User.groovy:43]: let's just add what is needed to support the additional unicode characters for now

**Comment by qualman** [comment]
No errors thrown in my testing as mentioned in the issue. Looks good!!

**Pull request merged**